go/client/cmd_team_add_members_bulk.go

```diff
@@ -0,0 +1,149 @@
+				Usage: "specify readers to add",
+			},
+			cli.StringFlag{
+				Name:  "w, writers",
+				Usage: "specify writers to add",
+			},
+			cli.StringFlag{
+				Name:  "a, admins",
+				Usage: "specify admins to add",
+			},
+			cli.StringFlag{
+				Name:  "o, owners",
+				Usage: "specify ownsers to add",
+			},
+			cli.BoolFlag{
+				Name:  "s, skip-chat-message",
+				Usage: "skip chat welcome message",
+			},
+		},
+		Description: teamAddMembersBulkDoc,
+	}
+}
+
+func NewCmdTeamAddMembersBulkRunner(g *libkb.GlobalContext) *CmdTeamAddMembersBulk {
+	return &CmdTeamAddMembersBulk{Contextified: libkb.NewContextified(g)}
+}
+
+func (c *CmdTeamAddMembersBulk) parseBulkList(v string, role keybase1.TeamRole) (n int, err error) {
+	if len(v) == 0 {
+		return 0, nil
+	}
+	parts := strings.Split(v, ",")
+	for _, p := range parts {
+		if len(p) == 0 {
+			return 0, errors.New("cannot specify an empty user")
+		}
+		c.arg.Users = append(c.arg.Users, keybase1.UserRolePair{AssertionOrEmail: p, Role: role})
+		n++
+	}
+	return n, nil
+
+}
+
+func (c *CmdTeamAddMembersBulk) ParseArgv(ctx *cli.Context) (err error) {
+	c.arg.Name, err = ParseOneTeamName(ctx)
+	if err != nil {
+		return err
+	}
+
+	var tot int
+
+	roles := []struct {
+		s string
+		t keybase1.TeamRole
+	}{
+		{"readers", keybase1.TeamRole_READER},
+		{"writers", keybase1.TeamRole_WRITER},
+		{"admins", keybase1.TeamRole_ADMIN},
+		{"owners", keybase1.TeamRole_OWNER},
+	}
+	for _, r := range roles {
+		n, err := c.parseBulkList(ctx.String(r.s), r.t)
+		if err != nil {
+			return err
+		}
+		tot += n
+	}
+	if tot == 0 {
+		return errors.New("Need at least one of --readers, --writers, --admins or --owners")
+	}
+	c.arg.SendChatNotification = !ctx.Bool("skip-chat-message")
+	return nil
+}
+
+func (c *CmdTeamAddMembersBulk) Run() error {
+	cli, err := GetTeamsClient(c.G())
+	if err != nil {
+		return err
+	}
+
+	err = cli.TeamAddMembersMultiRole(context.Background(), c.arg)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *CmdTeamAddMembersBulk) GetUsage() libkb.Usage {
+	return libkb.Usage{
+		Config:    true,
+		API:       true,
+		KbKeyring: true,
+	}
+}
+
+const teamAddMembersBulkDoc = `"keybase team add-members-bulk" allows you to add multiple users to a team, in bulk
+
+EXAMPLES:
+
+Add existing keybase users as writiers:
+
+    keybase team add-members-bulk acme --writers=alice,bob,charlie
+
+Add users via social assertions:
+
+    keybase team add-members-bulk acme --writers=alice+alice@github,bob@github,jerry@redder --readers=jon,bob32
+
+Add users via email:
+
+    keybase team add-members-bulk acme --readers='[max43@gmail.com]@email,[bill32@yahoo.com]@email' --writers='[lucy32@poems.com]@email'
+
+You can specify one or more of --readers, --writers, --admins, --owners, to add multiple
+roles at one go. For each of those lists, you can mix and match Keybase users, social
+assertions, and email addresses. Email addresses cannot be combined with other assertions,
+however.
```

Comment:
> Email addresses cannot be combined with other assertions, however.

Why is `[job@gmail.com]@email+job33` not allowed?

Reply:
we just don't handle it properly, i don't think.  And plus it could leak data, I'm not sure...